internal/services/postgres/postgresql_flexible_server_backup_resource.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/postgresql/2024-08-01/backups"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/locks"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/postgres/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ sdk.Resource = PostgresqlFlexibleServerBackupResource{}
+
+type PostgresqlFlexibleServerBackupResource struct{}
+
+func (r PostgresqlFlexibleServerBackupResource) ModelObject() interface{} {
+	return &PostgresqlFlexibleServerBackupResourceModel{}
+}
+
+type PostgresqlFlexibleServerBackupResourceModel struct {
+	Name     string `tfschema:"name"`
+	ServerId string `tfschema:"server_id"`
+}
+
+func (r PostgresqlFlexibleServerBackupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return backups.ValidateBackupID
+}
+
+func (r PostgresqlFlexibleServerBackupResource) ResourceType() string {
+	return "azurerm_postgresql_flexible_server_backup"
+}
+
+func (r PostgresqlFlexibleServerBackupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.FlexibleServerBackupName,
+		},
+
+		"server_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: backups.ValidateFlexibleServerID,
+		},
+	}
+}
+
+func (r PostgresqlFlexibleServerBackupResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
```

Comment:
Could you expose `completedTime` as an attribute?

Reply:
Updated